### ISSUE TITLE
Android TalkBack screen reader always reads CollectionView elements as clickable

### ISSUE THREAD

**Issue by FinnKr** [dotnet/maui#21700]
### Description

On Android any element in a CollectionView is read as clickable by the Android TalkBack screen reader.
This is the case even non clickable elements such as plain Labels.

### Steps to Reproduce

1. Create a CollectionView with a Label in its DataTemplate:
```xml
<CollectionView ItemsSource="{Binding Data}" SelectionMode="None">
    <CollectionView.ItemTemplate>
        <DataTemplate>
            <Label Text="{Binding IndexNr}"></Label>
        </DataTemplate>
    </CollectionView.ItemTemplate>
</CollectionView>
```
2. Run the app on an android device with TalkBack enabled
3. Navigate through the items with the screen reader

Result: "Double tap to activate" is announced on the labels of the collection view

Additional Infos:
* Inspecting the accessibility tree of the app shows that every element of the CollectionView is wrapped in a ViewGroup with the "click" action and "clickable" property assigned to it:
![Screenshot 2024-04-08 at 13 02 42](https://github.com/dotnet/maui/assets/49481876/82cc603d-3bb7-4a0e-8a08-3783bdd5d903)
 * On iOS the screen reader works as expected and does not read the elements as clickable

### Link to public reproduction project repository

https://github.com/FinnKr/MAUICollectionViewA11yAndroidBug

### Version with bug

8.0.14 SR3.1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 14

**Comment by ramonB1996** [comment]
This is also true for `ListView' afaik

**Comment by RoiChen001** [comment]
Can repro this issue at Android device on the latest 17.10.0 preview 2(8.0.10&8.0.14).

**Comment by AndreasDev3** [comment]
Has anyone found a solution to this? I just came across this issue in 9.0.40
